File: CodeTemplate/QuickSort.java
class QuickSort {

    public static void quickSort(int[] arr, int left, int right) {
        if (left >= right)
            return;
        int p = partition(arr, left, right);
        quickSort(arr, left, p - 1);
        quickSort(arr, p + 1, right);
    }

    private static int partition(int[] arr, int left, int right) {

        int pivot = right;
        int pos = left;
        for (int i = left; i < right; i++) {
            if (arr[i] < arr[pivot]) {
                swap(arr[i], arr[pos]);
                pos++;
            }
        }
        swap(arr[pos], arr[pivot]);
        return pos;
    }

    private static void swap(int a, int b) {
        int temp = a;
        a = b;
        b = temp;
    }
}

File: src/python/sample/QuickSort.py
def quicksort(arr):
    if len(arr) <= 1:return arr
    else:
        pivot = arr[len(arr)-1]
        left = [i for i in arr if i < pivot]
        right = [i for i in arr if i > pivot]
        return quicksort(left)+[pivot]+quicksort(right)

def main() -> None:
    arr = [3, 6, 8, 10, 1, 2, 1]
    print("Original array:", arr)
    sorted_arr = quicksort(arr)
    print("Sorted array:", sorted_arr)

main()